artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/QueueImpl.java

```diff
@@ -1567,7 +1565,16 @@ protected void finalize() throws Throwable {
 
    @Override
    public int getConsumerCount() {
-      return consumers.size();
+      int size = consumers.size();
+      // we don't want to count the redistributor, it is an internal transient entry in the consumer list
+      if (size > 0) {
+         synchronized (this) {
```

Comment:
-1 this method is hit alot on metric scraping it shouldnt be taking the lock else will cause perf issue, as metrics will be stealing the delivery lock.

Reply:
Maybe make the redistributor volatile and use a static atomic field updater to update and get its value

Reply:
fair enough... will sort

Reply:
implemented with the updater. thanks for the feedback :-)